stdlib/public/TensorFlow/Gradients.swift

```diff
@@ -220,30 +223,38 @@ extension Tensor where Scalar : TensorFlowFloatingPoint {
   ) -> (Tensor, (Tensor) -> (Tensor, Tensor)) {
     return (lhs - rhs, {
       [lhsShape = lhs.shapeTensor, rhsShape = rhs.shapeTensor] v in
-      return (v.unbroadcast(toShape: lhsShape),
-              -v.unbroadcast(toShape: rhsShape))
+      let (lhsAxes, rhsAxes) =
+        Raw.broadcastGradientArgs(s0: lhsShape, s1: rhsShape)
+      return (v.sum(squeezingAxes: lhsAxes).reshaped(toShape: lhsShape),
```

Comment:
I haven't looked too much, but I suspect that this extra reshape is not necessary. The lhsAxes should be sufficient to recover the original shape.

Reply:
That’s what I tried initially (more specifically, ‘sum(alongAxes:)’ but it didn’t work.

Reply:
@pschuh @rxwei the reshape is needed for handling dimensions with size 1. For example, say you do:
```swift
// x has shape [B, 5]
// y has shape [5]
// result has shape [B, 5]
let result = x + y
```
In this case, the broadcast indices for the gradient wrt to `y` will be `[0]` and so we’ll do something like:
```swift
let yGrad = seed.sum(alongAxes: [0]) // no reshape needed.
```
Now, let `y` have shape `[1, 5]`, which still broadcasts correctly for this example. The broadcast indices will now also be the same for the gradient (i.e., `[0]`). However, we need to do the reshape to recover the dimensions of size 1. Thus, the gradient needs to be computed as:
```swift
let yGrad = seed.sum(alongAxes: [0]).reshape(to: y.shape)
```
Having said that, I have a working implementation of these changes that I had made as part of a future `swift-apis` PR. I’ll try to open a PR here for this ASAP, but haven’t gotten the chance yet because I’m traveling to ICLR this week.

Reply:
To make sure we don't regress in the future, could you add a quick test case in your other PR to `swift-apis`? :-)

Reply:
Yeap, I will go ahead and add that. Given that the merge already happened, is it ok to make this change after we move stdlib to swift-apis? I'll update the two PRs doing the move tonight.

Reply:
Yes. Thanks!